lib/plugins/aws/lib/naming.js

```diff
@@ -208,10 +208,10 @@ module.exports = {
   // Websockets API
   getWebsocketsApiName() {
     if (
-      this.provider.serverless.service.provider.websocketsApiName &&
```

Comment:
We should still support the old syntax as well, but first check for the new one

Reply:
This is the only one I can't get to pass the tests. I used the code below. Any suggestions that might point me in the right direction?
getWebsocketsApiName() {
    if (this.provider.serverless.service.provider.websockets.apiName &&
      typeof this.provider.serverless.service.provider.websockets.apiName === 'string') {
        return `${this.provider.serverless.service.provider.websockets.apiName}`;
    } 
    if(this.provider.serverless.service.provider.websocketsApiName &&
      typeof this.provider.serverless.service.provider.websocketsApiName === 'string')
      {
        return `${this.provider.serverless.service.provider.websocketsApiName}`;
      }
    return `${this.provider.getStage()}-${this.provider.serverless.service.service}-websockets`;
  }

Reply:
This should also follow a similar approach with using `_.get` from `lodash`. We should not check the type of the property but rather only the fact that it exists, the schema validation checks that it's a string if its defined. 

Small request - when pasting the code samples, please format them all with code blocks as it's hard to read unformatted code. Thanks in advance 🙇

Reply:
The tests might be passing, but just glancing over the code I see that a lot of stuff listed in the issue is not addressed, I'm also pretty sure it's going to break in a few scenarios. Are you sure the PR in current form is ready for a deeper review?

Reply:
I appreciate your feedback. This is a learning process for me and I have identified several things I could have done better process wise and learned about lodash in the process. I'd like to address all the requirements for this issue. I see that I missed the deprecation.md. I assume that serverless._logDeprecation is a terminal command. I'll let you know when i resolve this and the other issues you mentioned.
Could you give me a hint of the break scenarios? Are they related to the lodash get?

Reply:
Hello @BillConley01 - no worries, this issue is actually a bit more complex that it might've seem at first - I'll try to provide more details to help you understand the scope better. 

For the potential breaking scenario, let's consider this check
```
getWebsocketsApiName() {
    if (_.get(this.provider.serverless.service.provider.websockets)) {
      return `${this.provider.serverless.service.provider.websockets.apiName}`;
    ....
}
```

In the check above, we're checking if the `this.provider.serverless.service.provider.websockets` is set, but that doesn't mean that `this.provider.serverless.service.provider.websockets.apiName` is set as well - we might have a situation where someone defined some other properties under `provider.websockets`, but didn't define `apiName` there - in such situation we should fallback to default value instead of returning an invalid string here.

---------

Another example we can find here:
```
Description: this.serverless.service.provider.websockets
            ? this.serverless.service.provider.websockets.description
            : this.serverless.service.provider.websocketsDescription || 'Serverless Websockets',
```

This is a part of the code that resolves the description. In the ternary here, we are checking if `provider.websockets` is defined - if it is, we return `provider.websocket.description`, but what if the `provider.websockets.description` is not defined? In that case, we would set it to `undefined`, where we would want to have the default of `Serverless Websockets` instead. 

Hope that helps a bit - if something is not clear, sorry about that and please let me know, I'll try to explain it a bit better 🙇

Reply:
Thanks for the explanation and I apologize for the delay. A family member passed away last week. I addressed the issues mentioned and updated the readme files. The only thing I am unsure of is where/how to add the deprecation code?

Reply:
Hey @BillConley01 - no worries about the delay and really sorry about your loss. 

As for the deprecation code, you can see how that can be implemented e.g. in this PR - https://github.com/serverless/serverless/pull/8277

If you have any additional questions, feel free to reach out 💯